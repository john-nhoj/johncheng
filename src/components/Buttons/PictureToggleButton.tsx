import classNames from 'classnames';
import React from 'react';

const PictureToggleButton = ({
  onClickHandler,
  label,
}: {
  onClickHandler: () => void;
  label: string;
}) => (
  <button className="relative m-2 uppercase" onClick={onClickHandler}>
    <div
      className={classNames(
        'absolute',
        'bg-red',
        'w-full',
        'h-full',
        'transform',
        '-skew-x-12',
        '-z-10'
      )}
    ></div>
    <span className={classNames('p-2')}>{label}</span>
  </button>
);

export { PictureToggleButton };
