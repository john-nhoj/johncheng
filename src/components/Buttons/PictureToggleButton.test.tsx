import React from 'react';
import renderer from 'react-test-renderer';
import { PictureToggleButton } from './PictureToggleButton';

test('Picture toggle button', () => {
  const header = renderer.create(
    <PictureToggleButton label="Toggle button" onClickHandler={() => {}} />
  );
  let tree = header.toJSON();
  expect(tree).toMatchSnapshot();
});
