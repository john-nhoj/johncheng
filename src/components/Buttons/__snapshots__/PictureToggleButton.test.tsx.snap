// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picture toggle button 1`] = `
<button
  className="relative m-2 uppercase"
  onClick={[Function]}
>
  <div
    className="absolute bg-red w-full h-full transform -skew-x-12 -z-10"
  />
  <span
    className="p-2"
  >
    Toggle button
  </span>
</button>
`;
